### PR TITLE
aktualizr-toml-update: Add variable to define the SOTA CLIENT systemd service name.

### PR DESCRIPTION
**Pull request by caiotpereira**
Add variable to define the SOTA CLIENT systemd service name.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,5 +32,13 @@ check: test
 	@test -x $(linter) || (echo "Please install linter from https://github.com/golangci/golangci-lint/releases/tag/v1.25.1 to $(HOME)/go/bin")
 	$(linter) run --build-tags vpn
 
+handlers:
```
Comment (mike-sul) [inline Makefile:35]: I think we could do this solely in yocto/meta-lmp.
I mean install `template-toml-update` to `${D}${datadir}/fioconfig/handlers` https://github.com/foundriesio/meta-lmp/blob/51caa20decd7cea2ceaeb169273ebcd089b45274/meta-lmp-base/recipes-support/fioconfig/fioconfig_git.bb#L34C70-L34C103 and then do the variable substitution after that.

Similar to what we do for the aklite service https://github.com/foundriesio/meta-lmp/blob/51caa20decd7cea2ceaeb169273ebcd089b45274/meta-lmp-base/recipes-sota/aktualizr/aktualizr_%25.bbappend#L40. The only difference is that we have to do it in some "do_postinstall" or "do_install_append".

Comment (caiotpereira) [reply Makefile:35]: @mike-sul , yes, it's possible, I was trying to keep the retro compatibility with the code.

Comment (caiotpereira) [reply Makefile:35]: I will revert the makefile changes and just keep the templates

File: Makefile [hunk 2]
```diff
@@ -30,3 +30,6 @@ check: test
 
 test:
 	go test ./... -v
+
+clean:
+	rm -f bin/fioconfig*
```
Comment (vkhoroz) [inline Makefile:35]: This is not related to your change. Please, make it a separate PR.
Also, there needs to be a line end at the last line (a matter of your IDE configuration).

File: README.md [hunk 3]
```diff
@@ -9,16 +9,20 @@ OTA devices communicate with a device-gateway using SSL client
...
 a fioconfig can extract this data to tmpfs (/var/run/secrets) so that
 they are only available at runtime.
 
 
 ## How to build
 `make bin/fioconfig-linux-amd64`
+<br />
+
 `make test`
+
```
Comment (vkhoroz) [inline README.md:27]: I appreciate grammar corrections to the readme.
Could you make them a separate commit/PR, so they are less distracting a topic of this change?

File: template/template-toml-update [hunk 4]
```diff
@@ -25,4 +25,4 @@ else
 	sed -e 's/^# MANAGED BY FIOCONFIG: //g' -i /var/sota/sota.toml
 fi
 
-systemctl restart aktualizr-lite
+systemctl restart @@SOTA_CLIENT@@
```
Comment (vkhoroz) [inline template/template-toml-update:28]: I have no strong opinion on the below, so I invite @mike-sul and @doanac to comment on this.

1. A comment on a file renaming operation:

It'd be nice to have a template like `contrib/aktualizr-toml-update.in`, so that it is clear to external developers of `fioconfig` what handlers are supported, without grepping the `meta-lmp-base`.
I mean, this project needs to be self-contained, as it is often being built and run outside LmP e.g. for development.

2. A comment on the templating itself:

I'd prefer if the `SOTA_CLIENT` goes from environment variable, similarly to the `CONFIG_FILE`.
Although, I'd give it the default value like this `${SOTA_CLIENT:-aktualizr-lite}` (so that no env variable equals "aktualizr-lite").
A reason is the same as above: we need an ability to build and run `fioconfig` outside LmP.

There are several ways how this environment variable can get into here:
- through templating of the `interrnal/app.go` at https://github.com/foundriesio/fioconfig/blob/main/internal/app.go#L303 in bb-append before `go build` (this is what I like the least).
- leveraging the `/etc/default/fioconfig` possibility at https://github.com/foundriesio/meta-lmp/blob/main/meta-lmp-base/recipes-support/fioconfig/fioconfig/fioconfig.service#L7 (added by @munoz0raul recently).
- adding the SOTA_CLIENT environment variable to `fioconfig/fioconfig.service` in bb-append (similar to above).

The latter two options make both `fioconfig` and `meta-lmp/.../fioconfig` fully encapsulated, but still provide the necessary level of integration and flexibilty.

Comment (doanac) [reply template/template-toml-update:28]: changing the value the systemd service name from immutable (ostree) to an environment variable may have security ramifications. This templating approach might be more "secure".

Comment (mike-sul) [reply template/template-toml-update:28]: The original version that Caio posted was kind of "self-contained" (a new Makefile target that turns the template to actual value).

I don't like this approach because the update script customization is Yocto/meta-lmp specific, it's not needed for the fioconfig project at all.

Relying on the env variable is not a good option because the sota client is defined at build time, so the toml update script should be defined at build time and it's behavior should not be customizable at runtime. Otherwise, customers can easily shoot themself into the foot.

Ideally, I would make this customization (which of the custom client to restart in the `aktualizr-toml-update`) solely within `meta-lmp` no any changes to fioconfig.

Maybe, we can make the fioconfig recipe replace `systemctl restart aktualizr-lite` with `systemctl restart ${SOTA_CLIENT}` so no need to make any changes to fioconfig project at all?

Comment (mike-sul) [reply template/template-toml-update:28]: I would do in `do_configure:append()`

```
sed -e 's|systemctl restart aktualizr-lite|systemctl restart ${SOTA_CLIENT}|g' \
        ${S}/src/${GO_IMPORT}/template/template-toml-update > ${S}/src/${GO_IMPORT}/template/template-toml-update-customized
```

And then in `do_install:append()`

```
install -m 0755 ${S}/src/${GO_IMPORT}/contrib/aktualizr-toml-update-customized ${D}${datadir}/fioconfig/handlers
```

Since the fioconfig is linked/bound with specific version of `fioconfig` (`SRCREV=`) I don't see any issues with "sedding" of `systemctl restart aktualizr-lite`.

Comment (vkhoroz) [reply template/template-toml-update:28]: @doanac @mike-sul I buy your arguments; that's why I told that I have no strong opinion.

Actually, I like the latest suggestion from @mike-sul, which is to replace the whole thing.
I _believe_ we can improve on it by declaring a local script variable:
```
SOTA_CLIENT=aktualizr-lite

systemctl restart ${SOTA_CLIENT}
```

This way, the bb-append script can do a more intuitive substitution:
```
sed -e "s|SOTA_CLIENT=aktualizr-lite|SOTA_CLIENT=${SOTA_CLIENT}|g" -i ${S}/src/${GO_IMPORT}/template/template-toml-update > > ${S}/src/${GO_IMPORT}/template/template-toml-update-customized
```
It is also forward-compatible, as it allows to reuse the SOTA_CLIENT in all _future_ places inside a script where we need it.

Comment (mike-sul) [reply template/template-toml-update:28]: The improvement makes sense to me.

Comment (caiotpereira) [reply template/template-toml-update:28]: @mike-sul , nice, this is more simple.

File: README.md (hunk 3 above)
Comment (caiotpereira) [reply README.md:27]: Ok

**Comment by vkhoroz** [comment]
@caiotpereira the last commit is good.

Please, squash all commits, force-push, and then I will merge.

File: contrib/aktualizr-toml-update [hunk 5]
```diff
@@ -1,5 +1,8 @@
 #!/bin/sh -e
 
+#Defines the SOTA Client systemd service name.
```
Comment (vkhoroz) [inline contrib/aktualizr-toml-update:3]: I believe we don't need this comment. The variable name is self-descriptive.

Comment (caiotpereira) [reply contrib/aktualizr-toml-update:3]: @vkhoroz I was testing one idea but didn't work. I was trying to simplify and avoid to create a template file but didn't work.
I was using `sed -i `  to avoid copy the file but this is not a good idea. On yocto side if someone is running a local build and change the `SOTA_CLIENT` after the first build it will not update.

Comment (caiotpereira) [reply contrib/aktualizr-toml-update]: indeed. you are correcty. Thank you for the clarification.

**Comment by mike-sul** [comment]
@caiotpereira  Great work! Thanks!
